### PR TITLE
[Timob-18713] Kitchen Sink using new Facebook module

### PR DESCRIPTION
**Pull request by ashcoding**
Jira: https://jira.appcelerator.org/browse/TIMOB-18713

### CODE REVIEW

**Comment by ashcoding** [comment]
Guide to test:
Download latest release of FB module for Android and iOS from here. https://github.com/appcelerator-modules/ti.facebook/releases

Then run kitchen sink app with those modules.

File: Resources/ui/common/mashups/facebook_login_logout.js [hunk 1]
```diff
@@ -1,62 +1,146 @@
...
 	});
 	win.add(label);
+		
+	fb.addEventListener('login',function(e) {
+		// You *will* get this event if loggedIn == false below
+     	// Make sure to handle all possible cases of this event
+     	if (e.success) {
+ 			alert('login from uid: '+e.uid+', name: '+JSON.parse(e.data).name);
+ 			label.text = 'Logged In = ' + fb.loggedIn;
+     	}
```
Comment (cheekiatng) [inline Resources/ui/common/mashups/facebook_login_logout.js]: note spacing and indentation

Comment (ashcoding) [inline Resources/ui/common/mashups/facebook_login_logout.js]: Okay.

File: Resources/ui/common/mashups/facebook_photos.js [hunk 2]
```diff
@@ -78,7 +64,23 @@ function fb_photos(_args) {
 			{
 				b1.title = 'Uploading Photo...';
 				var data = {picture: event.media};
-				facebook.requestWithGraphPath('me/photos', data, "POST", showRequestResult);
+				//If publish_actions permission is not granted, request it
+				if(fb.permissions.indexOf('publish_actions') < 0) {
+					fb.requestNewPublishPermissions(['publish_actions'],fb.AUDIENCE_FRIENDS,function(e){
+						if(e.success) {
+							Ti.API.info('Permissions:'+fb.permissions);
+							fb.requestWithGraphPath('me/photos', data, "POST", showRequestResult);
```
Comment (cheekiatng) [inline Resources/ui/common/mashups/facebook_photos.js]: not working in iOS simulator. maybe try {source: data} instead.

File: Resources/ui/common/mashups/facebook_photos.js [hunk 3]
```diff
@@ -78,7 +64,23 @@ function fb_photos(_args) {
...
+						}
+						if(e.error) {
+							Ti.API.info('Publish permission error');
+						}
+						if(e.cancelled) {
+							Ti.API.info('Publish permission cancelled');
+						}
+					});
+				} else {
+					fb.requestWithGraphPath('me/photos', data, "POST", showRequestResult);
```
Comment (cheekiatng) [inline Resources/ui/common/mashups/facebook_photos.js]: same as above comment. see Facebook Graph API reference.

File: Resources/ui/common/mashups/facebook_photos.js [hunk 4]
```diff
@@ -98,11 +101,32 @@ function fb_photos(_args) {
 			caption: 'behold, a flower',
 			picture: blob
 		};
-		facebook.request('photos.upload', data, showRequestResult);
+		//If publish_actions permission is not granted, request it
+		if(fb.permissions.indexOf('publish_actions') < 0) {
+			fb.requestNewPublishPermissions(['publish_actions'],fb.AUDIENCE_FRIENDS,function(e){
+				if(e.success) {
+					Ti.API.info('Permissions:'+fb.permissions);
+					fb.requestWithGraphPath('me/photos', data, "POST", showRequestResult);
```
Comment (cheekiatng) [inline Resources/ui/common/mashups/facebook_photos.js]: this one uses {url: data} i think.

File: Resources/ui/common/mashups/facebook_photos.js [hunk 5]
```diff
@@ -98,11 +101,32 @@ function fb_photos(_args) {
...
+				}
+				if(e.error) {
+					Ti.API.info('Publish permission error');
+				}
+				if(e.cancelled) {
+					Ti.API.info('Publish permission cancelled');
+				}
+			});
+		} else {
+			fb.requestWithGraphPath('me/photos', data, "POST", showRequestResult);
```
Comment (cheekiatng) [inline Resources/ui/common/mashups/facebook_photos.js]: same as above.

File: Resources/ui/common/mashups/facebook_photos.js [hunk 6]
```diff
@@ -78,7 +64,23 @@ function fb_photos(_args) {
 			{
 				b1.title = 'Uploading Photo...';
 				var data = {picture: event.media};
```
Comment (cheekiatng) [inline Resources/ui/common/mashups/facebook_photos.js]: not working. refer to Facebook graph api ref. maybe something like var data = {source: event.media}

File: Resources/ui/common/mashups/facebook_photos.js [hunk 7]
```diff
@@ -98,11 +101,32 @@ function fb_photos(_args) {
 			caption: 'behold, a flower',
 			picture: blob
```
Comment (cheekiatng) [inline Resources/ui/common/mashups/facebook_photos.js]: not working. consider using {message: 'behold, a flower', url: blob} as according to Facebook graph api ref.

**Comment by ashcoding** [comment]
The publishing photo for android seems to be working. Not working for the iOS portion. Currently investigating this. Trying to see if it is fb permissions or something that's preventing it from publishing a picture on the graph api.

**Comment by cheekiatng** [comment]
CR and FT passed. Merging

**Pull request merged**